signature-type.go

```diff
@@ -27,6 +27,8 @@ const (
 	SignatureV4Streaming
 )
 
+var emptySHA256 = sum256([]byte{})
```

Comment:
why is `[]byte{}` used in favour of `nil`? The default value for a slice is `nil` so we can avoid setting `metadata.contentSHA256Bytes` like we are doing in this PR.